app/features/receive/cashu-token-swap-service.ts

```diff
@@ -103,15 +103,26 @@ export class CashuTokenSwapService {
       tokenSwap.outputAmounts,
     );
 
-    const newProofs = await this.swapProofs(wallet, tokenSwap, outputData);
-    const allProofs = [...account.proofs, ...newProofs];
-
-    await this.tokenSwapRepository.completeTokenSwap({
-      tokenHash: tokenSwap.tokenHash,
-      swapVersion: tokenSwap.version,
-      proofs: allProofs,
-      accountVersion: account.version,
-    });
+    try {
+      const newProofs = await this.swapProofs(wallet, tokenSwap, outputData);
+      const allProofs = [...account.proofs, ...newProofs];
+
+      await this.tokenSwapRepository.completeTokenSwap({
+        tokenHash: tokenSwap.tokenHash,
+        swapVersion: tokenSwap.version,
+        proofs: allProofs,
+        accountVersion: account.version,
+      });
+    } catch (error) {
+      if (error instanceof Error && error.message === 'TOKEN_ALREADY_CLAIMED') {
+        await this.tokenSwapRepository.fail({
+          tokenHash: tokenSwap.tokenHash,
+          version: tokenSwap.version,
+          reason: 'Token already claimed',
+        });
+      }
+      throw error;
```

Comment:
won't throwing error even in case of already claim trigger retries for that too?